bfs_battle_for_supremacy/game_logic/managers/map_manager.py

```diff
@@ -1,9 +1,135 @@
 from bfs_battle_for_supremacy.game_logic.entities.map import Map
 from bfs_battle_for_supremacy.game_logic.entities.square import Square
+from collections import deque
 
 
 class MapManager:
     game_map = Map(10, 10)
+    movement_counter = 20
+
+    current_position = None
+    current_object = None
+    target_position = None
+    selection_counter = 0
+
+    @staticmethod
+    def reset_movement_counter():
+        MapManager.movement_counter = 20
+
+    @staticmethod
+    def reset_selection():
+        MapManager.current_position = None
+        MapManager.current_object = None
+        MapManager.target_position = None
+        MapManager.selection_counter = 0
+
+    @staticmethod
+    def select_square(square: Square, current_player):
+        if MapManager.selection_counter == 0:
+            if not square.is_empty:
+                content = square.get_content()
+                if (
+                    content == current_player
+                    or content in current_player.monsters
+                ):
+                    MapManager.current_position = square
+                    MapManager.current_object = content
+                    MapManager.selection_counter = 1
+                else:
+                    print("Invalid selection. Please select a valid object.")
+            else:
+                print("Invalid selection. Square is empty.")
+
+        elif MapManager.selection_counter == 1:
+            if MapManager.check_availability(square):
+                MapManager.target_position = square
+                MapManager.selection_counter = 2
+                print(
+                    "Selected target position:"
+                    + f"({square.row}, {square.column})"
+                )
+                MapManager.move_with_bfs(current_player)
+            else:
+                print(
+                    "Target position is not empty."
+                    + "Please select an empty square."
+                )
+
+        else:
+            print("Too many selections. Resetting selection.")
+            MapManager.reset_selection()
```

Comment:
don't reset selection here (if the player selected some square while his monster is walking this selecting should be totally ignored)